subsys/bluetooth/host/hci_core.c

```diff
@@ -1510,8 +1510,12 @@ void bt_hci_le_enh_conn_complete_sync(struct bt_hci_evt_le_enh_conn_complete_v2
 
 	bt_addr_le_copy(&conn->le.init_addr, &peer_addr);
 
-	/* There is no random addr to get, set responder addr to local identity addr. */
-	bt_addr_le_copy(&conn->le.resp_addr, &bt_dev.id_addr[conn->id]);
+	if (IS_ENABLED(CONFIG_BT_PRIVACY)) {
```

Comment:
Before we introduce additional checks like these, we need to make sure that privacy is enabled in the controller, otherwise `evt->local_rpa` will not be valid. `CONFIG_BT_PRIVACY` enabled privacy in the host, but in split builds that does not ensure that the controller will support privacy. We likely need something like:
`CONFIG_BT_PRIVACY && BT_FEAT_LE_PRIVACY()`.

@hermabe can we start by creating a GitHub issue for this?

Reply:
Created https://github.com/zephyrproject-rtos/zephyr/issues/59653.

I will be OoO for the next 3 weeks so will not be able to work on it in that time.